ls1intum/Artemis
# Add new programming exercise for Swift

### Checklist

### Motivation and Context


We want to have Swift exercises in Artemis, so that it also might be used in the iPraktikum.

### Description

This PR adds a template for Swift programming exercises but doesn't contain valid tests for the CI, yet.
A regex pattern for Swift package names was added, so that only unseparated names are allowed, such as "SwiftEx1" and not "Swift_Ex.1"

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create a new Programming Exercise, select Swift in the dropdown menu and enter a package name
4. Verify it was successful and start the exercise
4. Click on open in Online Editor 
5. See a working template for BubbleSort and MergeSort as a Swift code exercise which package's name is the one defined in step 3.
6. Verify that the ACE module could be loaded and no 404 error appears in the console.

Note: The new programming languages are only supported with Bamboo, so make sure to use TS1 or TS3 when testing.

### Test Coverage

The test coverage is almost staying the same, because only existing files were edited, which already have parameterized tests like the Programming Language Enum.

### Screenshots


![image](https://user-images.githubusercontent.com/33484318/98237527-51d4f180-1f65-11eb-8598-90980042f5e1.png)

![image](https://user-images.githubusercontent.com/33484318/98237564-62856780-1f65-11eb-912b-eb0a8c33708d.png)

![image](https://user-images.githubusercontent.com/33484318/98237676-8c3e8e80-1f65-11eb-9838-0e464ec0556c.png)

![image](https://user-images.githubusercontent.com/33484318/98237778-ba23d300-1f65-11eb-982f-dfce93f0bd20.png)
